terminal/bundles/org.eclipse.terminal.view.ui/src/org/eclipse/terminal/view/ui/internal/handler/LaunchTerminalCommandHandler.java

```diff
@@ -130,8 +134,20 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
 				executeDelegate(selection, delegate);
 			}
 		}
+	}
 
-		return null;
+	/**
+	 * Determines the command ID based on the active part context.
+	 * In E4, we can't directly access the command ID from the event, so we use
+	 * a default behavior.
```

Comment:
@vogella @opcoach @ruspl-afed is this correct? If not what would be the E4 way of doing this? If yes, please open an issue and link it here so we can keep track of missing E4 functionality.